### PR TITLE
[JN-1425] withdrawn reason

### PR DESCRIPTION
**Pull request by devonbush**
#### DESCRIPTION (include screenshots, and mobile screenshots for participant UX)

Allows specifying a withdrawn reason and note in the UX.  This reason is auto-set to DUPLICATE for merges

image

Along the way, this upgrades the enrolleeExtService with annotations

#### TO TEST:  *(simple manual steps for confirming core behavior -- used for pre-release checks)*

1. redeploy apiAdminApp
2. go to `https://localhost:3000/demo/studies/heartdemo/env/sandbox/participants/HDBASC/withdrawal`
3. confirm you can set a note and reason for the withdrawal
4. wthdraw the participant
5. goto `https://localhost:3000/demo/studies/heartdemo/env/sandbox/participants/withdrawn`
6. confirm you see the withdrawal reason and note

### CODE REVIEW

**Comment by devonbush** [comment]
ugh, no way to get a clean version of this -- will wait until other stuff gets merged

File: core/src/main/resources/db/changelog/changesets/2024_10_17_withdrawal_reason.yaml
Comment (MatthewBemis) [inline core/src/main/resources/db/changelog/changesets/2024_10_17_withdrawal_reason.yaml]:
```suggestion
      id: "withdrawal_reason"
```

I assume this is copy/paste leftover- this id is a duplicate of a past changeset

File: core/src/main/resources/db/changelog/changesets/2024_10_17_withdrawal_reason.yaml
Comment (MatthewBemis) [inline core/src/main/resources/db/changelog/changesets/2024_10_17_withdrawal_reason.yaml]: nit: I think this one should go in a separate changeset since it's unrelated to the rest of the changes

**Review by MatthewBemis — APPROVED** [review]
Looks great, just a couple of small suggestions

**Review by connorlbark — APPROVED** [review]

**Pull request merged**
